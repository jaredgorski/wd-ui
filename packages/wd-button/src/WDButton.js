import Component from 'metal-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';
import './WDButton.scss';

import templates from './WDButton.soy.js';

/**
 * WDButton component.
 */
class WDButton extends Component {}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
WDButton.STATE = {
  /**
   * ID to be applied to the element.
   * @type {!String}
   * @default example
   */
  id: Config.string().value('example'),

  /**
   * @type {?boolean}
   * @default false
   */
  disabled: Config.bool().value(false),

  /**
   * @type {?(string|undefined)}
   * @default undefined
   */
  elementClasses: Config.string(),

  /**
   * @type {?string}
   * @default 'primary'
   */
  style: Config.oneOf(['primary', 'outline', 'ghost', 'danger', 'destructive', 'float']).value('primary'),

  /**
   * @type {?string}
   * @default 'accent'
   */
  color: Config.oneOf(['accent', 'black', 'green', 'blue-light', 'blue', 'violet', 'red', 'orange', 'yellow']).value('accent'),

  /**
   * @type {?string}
   * @default 'm'
   */
  size: Config.oneOf(['xs', 's', 'm', 'l', 'xl', 'xxl']).value('s'),

  /**
   * @type {?string|undefined}
   * @default undefined
   */
  width: Config.string(),

  /**
   * @type {?boolean}
   * @default false
   */
  hardWidth: Config.bool().value(false),

  /**
   * @type {?string}
   * @default undefined
   */
  weight: Config.oneOf(['lightweight', 'middleweight', 'heavyweight']),

  /**
   * @type {?boolean}
   * @default undefined
   */
  darkTheme: Config.bool(),

  /**
   * @type {?(string|undefined)}
   * @default undefined
   */
  borderRadius: Config.oneOf(['borderRadius-6px', 'rounded']),

  /**
   * @type {?boolean}
   * @default false
   */
  group: Config.bool().value(false),

  /**
   * @type {?string}
   * @default 'default'
   */
  groupType: Config.oneOf(['default', 'segmented']).value('default'),

  /**
   * @type {?string}
   * @default '0px'
   */
  groupSpacing: Config.string().value('0px'),

  /**
   * @type {?Array}
   * @default empty
   */
  groupButtons: Config.arrayOf(
    Config.shapeOf({
      disabled: Config.bool().value(false),
      elementClasses: Config.string(),
      style: Config.oneOf(['primary', 'outline', 'ghost', 'danger', 'destructive', 'float']).value('primary'),
      color: Config.oneOf(['accent', 'black', 'green', 'blue-light', 'blue', 'violet', 'red', 'orange', 'yellow']).value('accent'),
      size: Config.oneOf(['xs', 's', 'm', 'l', 'xl', 'xxl']).value('s'),
      width: Config.string(),
      hardWidth: Config.bool().value(false),
      weight: Config.oneOf(['lightweight', 'middleweight', 'heavyweight']),
      darkTheme: Config.bool(),
      href: Config.string(),
      icon: Config.any(),
      classIcon: Config.string(),
      iconAlignment: Config.oneOf(['right-center', 'left-center', 'right-justified', 'left-justified']),
      id: Config.string(),
      label: Config.any(),
      name: Config.string(),
      rel: Config.string(),
      target: Config.oneOf(['_blank', '_self', '_parent', '_top']),
      type: Config.oneOf(['button', 'reset', 'submit']).value('button'),
      value: Config.string(),
    })
  ).value([{}]),

  /**
   * @type {?(string|undefined)}
   * @default undefined
   */
  href: Config.string(),

  /**
   * @type {?(html|string|undefined)}
   * @default undefined
   */
  icon: Config.any(),

  /**
   * @type {?(string|undefined)}
   * @default undefined
   */
  classIcon: Config.string(),

  /**
   * @type {?(string|undefined)}
   * @default undefined
   */
  iconAlignment: Config.oneOf(['right-center', 'left-center', 'right-justified', 'left-justified']),

  /**
   * @type {?(string|undefined)}
   * @default undefined
   */
  id: Config.string(),

  /**
   * @type {?(html|string|undefined)}
   * @default undefined
   */
  label: Config.any(),

  /**
   * @type {?(string|undefined)}
   * @default undefined
   */
  name: Config.string(),

  /**
   * @type {?(string|undefined)}
   * @default undefined
   */
  rel: Config.string(),

  /**
   * @type {?(string|undefined)}
   * @default undefined
   */
  target: Config.oneOf(['_blank', '_self', '_parent', '_top']),

  /**
   * @type {?string}
   * @default 'button'
   */
  type: Config.oneOf(['button', 'reset', 'submit']).value('button'),

  /**
   * @type {?(string|undefined)}
   * @default undefined
   */
  value: Config.string(),
};

Soy.register(WDButton, templates);

export {WDButton};
export default WDButton;
