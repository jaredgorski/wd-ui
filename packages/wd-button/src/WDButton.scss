@import "wd-base/build/main";


$font-primary: "Circular", sans-serif;
$accent: #00ae55;


$btn-accent: $accent;
$btn-black: #0e141a;
$btn-green: #00ae55;
$btn-blue-light: #00a4ff;
$btn-blue: #1f7aff;
$btn-violet: #9552ef;
$btn-red: #ff4040;
$btn-orange: #ff6e02;
$btn-yellow: #ffb700;

/* ==========================================================================
   Buttons
   ========================================================================== */
.btn {
  box-sizing: border-box;
  align-content: center;
  align-items: center;
  display: inline-flex;
  border: 0 solid transparent;
  border-radius: 4px;
  font-family: $font-primary;
  justify-content: center;
  text-align: center;
  white-space: nowrap;
  transition: 0.33s;
  cursor: pointer;

  &:active,
  &:focus {
    outline: 0;
  }

  &[disabled] {
    cursor: not-allowed;
    pointer-events: none;
    text-decoration: none;
  }

  .icon svg {
    transition: 0.33s;
  }
}

/* Style
   ========================================================================== */
@mixin button-primary($overlay-color, $btn-color) {
  background-color: $btn-color;
  box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.18), 0 0 0 0.65px $btn-color;
  color: rgba(255, 255, 255, 0.98);

  .icon svg {
    fill: rgba(255, 255, 255, 0.98);
  }

  &:hover {
    background-color: mix($overlay-color, $btn-color, 10%);
    box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.2), 0 0 0 0.65px $btn-color;
  }

  &:focus {
    background-color: mix($overlay-color, $btn-color, 20%);
    box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.2), 0 0 0 0.65px $btn-color, 0 0 0 4px rgba($btn-color, 0.3);
  }

  &:active {
    background-color: mix($overlay-color, $btn-color, 20%);
    box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.3), 0 0 0 0.65px $btn-color;
  }

  &.btn-selected {
    background-color: mix($overlay-color, $btn-color, 30%);
    box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.5), 0 0 0 0.65px $btn-color;
  }

  &[disabled] {
    background-color: rgba($btn-color, 0.4);
    box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.1), 0 0 0 0.65px rgba($btn-color, 0.4);
    color: rgba(255, 255, 255, 0.3);

    .icon svg {
      fill: rgba(255, 255, 255, 0.3);
    }
  }
}

@mixin button-outline($btn-color) {
  background-color: rgba(14, 20, 26, 0.02);
  box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.18);
  color: rgba(14, 20, 26, 0.8);

  .icon svg {
    fill: rgba(14, 20, 26, 0.8);
  }

  &:hover {
    background-color: rgba(14, 20, 26, 0.03);
    box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.3);

    &.btn-darktheme {
      background-color: rgba(255, 255, 255, 0.03);
      box-shadow: 0 0 0 0.65px rgba(255, 255, 255, 0.3);
    }
  }

  &:focus {
    background-color: rgba(14, 20, 26, 0.03);
    box-shadow: 0 0 0 3px rgba($btn-color, 0.3), 0 0 0 0.65px rgba(14, 20, 26, 0.3);

    &.btn-darktheme {
      background-color: rgba(255, 255, 255, 0.03);
      box-shadow: 0 0 0 3px rgba($btn-color, 0.3), 0 0 0 0.65px rgba(255, 255, 255, 0.3);
    }
  }

  &:active {
    background-color: rgba(14, 20, 26, 0.06);
    box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.3);

    &.btn-darktheme {
      background-color: rgba(255, 255, 255, 0.06);
      box-shadow: 0 0 0 0.65px rgba(255, 255, 255, 0.3);
    }
  }

  &.btn-selected {
    background-color: rgba($btn-color, 0.06);
    box-shadow: 0 0 0 0.65px $btn-color;
    color: $btn-color;

    .icon svg {
      fill: $btn-color;
    }

    &.btn-darktheme {
      background-color: rgba($btn-color, 0.06);
      box-shadow: 0 0 0 0.65px $btn-color;
    }
  }

  &[disabled] {
    background-color: rgba(14, 20, 26, 0.06);
    box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.12);
    color: rgba(14, 20, 26, 0.3);

    .icon svg {
      fill: rgba(14, 20, 26, 0.3);
    }

    &.btn-darktheme {
      background-color: rgba(255, 255, 255, 0.06);
      box-shadow: 0 0 0 0.65px rgba(255, 255, 255, 0.12);
      color: rgba(255, 255, 255, 0.3);

      .icon svg {
        fill: rgba(255, 255, 255, 0.3);
      }
    }
  }

  &.btn-darktheme {
    background-color: rgba(255, 255, 255, 0.02);
    box-shadow: 0 0 0 0.65px rgba(255, 255, 255, 0.18);
    color: rgba(255, 255, 255, 0.98);

    .icon svg {
      fill: rgba(255, 255, 255, 0.98);
    }
  }
}

@mixin button-ghost($btn-color) {
  background-color: transparent;
  color: rgba(14, 20, 26, 0.8);

  .icon svg {
    fill: rgba(14, 20, 26, 0.8);
  }

  &:hover {
    background-color: rgba(14, 20, 26, 0.08);

    &.btn-darktheme {
      background-color: rgba(255, 255, 255, 0.08);
    }
  }

  &:focus {
    background-color: rgba($btn-color, 0.1);
    box-shadow: 0 0 0 3px rgba($btn-color, 0.3);
  }

  &:active {
    background-color: rgba($btn-color, 0.1);
  }

  &.btn-selected {
    background-color: rgba($btn-color, 0.2);
    color: $btn-color;

    .icon svg {
      fill: $btn-color;
    }

    &.btn-darktheme {
      background-color: rgba($btn-color, 0.2);
    }
  }

  &[disabled] {
    color: rgba(14, 20, 26, 0.3);

    .icon svg {
      fill: rgba(14, 20, 26, 0.3);
    }

    &.btn-darktheme {
      color: rgba(255, 255, 255, 0.3);

      .icon svg {
        fill: rgba(255, 255, 255, 0.3);
      }
    }
  }

  &.btn-darktheme {
    color: rgba(255, 255, 255, 0.98);

    .icon svg {
      fill: rgba(255, 255, 255, 0.98);
    }
  }
}

@mixin button-float($btn-color) {
  background-color: #fff;
  box-shadow: 0 1px 3px 0 rgba(14, 20, 26, 0.06), 0 6px 12px -3px rgba(14, 20, 26, 0.18);
  color: rgba(14, 20, 26, 0.8);

  .icon svg {
    fill: rgba(14, 20, 26, 0.8);
  }

  &:hover {
    box-shadow: 0 1px 2px 0 rgba(14, 20, 26, 0.12), 0 9px 15px -1px rgba(14, 20, 26, 0.24);
    color: $btn-color;

    .icon svg {
      fill: $btn-color;
    }

    &.btn-darktheme {
      box-shadow: 0 1px 2px 0 rgba(14, 20, 26, 0.12), 0 9px 15px -1px rgba(14, 20, 26, 0.24);
      color: $btn-color;
    }
  }

  &:focus {
    box-shadow: 0 1px 2px 0 rgba(14, 20, 26, 0.12), 0 9px 15px -1px rgba(14, 20, 26, 0.24), 0 0 0 3px rgba($btn-color, 0.3);
    color: $btn-color;

    .icon svg {
      fill: $btn-color;
    }
  }

  &:active {
    box-shadow: 0 1px 2px 0 rgba(14, 20, 26, 0.12), 0 12px 18px -4px rgba(14, 20, 26, 0.24);
    color: $btn-color;

    .icon svg {
      fill: $btn-color;
    }
  }

  &.btn-selected {
    box-shadow: 0 1px 2px 0 rgba(14, 20, 26, 0.12), 0 9px 18px -2px rgba(14, 20, 26, 0.24), 0 0 0 0.65px rgba($btn-color, 0.3);
    color: rgba(14, 20, 26, 0.8);

    .icon svg {
      fill: rgba(14, 20, 26, 0.8);
    }

    &.btn-darktheme {
      box-shadow: 0 1px 2px 0 rgba(14, 20, 26, 0.12), 0 9px 18px -2px rgba(14, 20, 26, 0.24), 0 0 0 0.65px rgba($btn-color, 0.3);
      color: rgba(255, 255, 255, 0.98);
    }
  }

  &[disabled] {
    box-shadow: 0 1px 3px 0 rgba(14, 20, 26, 0.06), 0 6px 12px -3px rgba(14, 20, 26, 0.18);
    color: rgba(14, 20, 26, 0.3);

    .icon svg {
      fill: rgba(14, 20, 26, 0.3);
    }

    &.btn-darktheme {
      color: rgba(255, 255, 255, 0.3);

      .icon svg {
        fill: rgba(255, 255, 255, 0.3);
      }
    }
  }

  &.btn-darktheme {
    background-color: #0e141a;
    color: rgba(255, 255, 255, 0.98);

    .icon svg {
      fill: rgba(255, 255, 255, 0.98);
    }
  }
}

.btn.btn-destructive {
  background-color: #ff4040;
  box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.18), 0 0 0 0.65px #ff4040;
  color: rgba(255, 255, 255, 0.98);

  .icon svg {
    fill: rgba(255, 255, 255, 0.98);
  }

  &:hover {
    background-color: mix(#0e141a, #ff4040, 10%);
    box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.2), 0 0 0 0.65px #ff4040;
  }

  &:focus {
    background-color: mix(#0e141a, #ff4040, 20%);
    box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.2), 0 0 0 0.65px #ff4040, 0 0 0 4px rgba(255, 64, 64, 0.3);
  }

  &:active {
    background-color: mix(#0e141a, #ff4040, 20%);
    box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.3), 0 0 0 0.65px #ff4040;
  }

  &.btn-selected {
    background-color: mix(#0e141a, #ff4040, 30%);
    box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.5), 0 0 0 0.65px #ff4040;
  }

  &[disabled] {
    background-color: rgba(255, 64, 64, 0.4);
    box-shadow: 0 0 0 0.65px rgba(14, 20, 26, 0.1), 0 0 0 0.65px rgba(255, 64, 64, 0.4);
    color: rgba(255, 255, 255, 0.3);

    .icon svg {
      fill: rgba(255, 255, 255, 0.3);
    }
  }
}

/* Color
   ========================================================================== */
.btn {
  &.btn-primary-accent {
    @include button-primary(#0e141a, $btn-accent);
  }
  &.btn-primary-black {
    @include button-primary(#fff, $btn-black);
  }
  &.btn-primary-green {
    @include button-primary(#0e141a, $btn-green);
  }
  &.btn-primary-blue-light {
    @include button-primary(#0e141a, $btn-blue-light);
  }
  &.btn-primary-blue {
    @include button-primary(#0e141a, $btn-blue);
  }
  &.btn-primary-violet {
    @include button-primary(#0e141a, $btn-violet);
  }
  &.btn-primary-red {
    @include button-primary(#0e141a, $btn-red);
  }
  &.btn-primary-orange {
    @include button-primary(#0e141a, $btn-orange);
  }
  &.btn-primary-yellow {
    @include button-primary(#0e141a, $btn-yellow);
  }

  &.btn-outline-accent {
    @include button-outline($btn-accent);
  }
  &.btn-outline-black {
    @include button-outline($btn-black);
  }
  &.btn-outline-green {
    @include button-outline($btn-green);
  }
  &.btn-outline-blue-light {
    @include button-outline($btn-blue-light);
  }
  &.btn-outline-blue {
    @include button-outline($btn-blue);
  }
  &.btn-outline-violet {
    @include button-outline($btn-violet);
  }
  &.btn-outline-red {
    @include button-outline($btn-red);
  }
  &.btn-outline-orange {
    @include button-outline($btn-orange);
  }
  &.btn-outline-yellow {
    @include button-outline($btn-yellow);
  }

  &.btn-ghost-accent {
    @include button-ghost($btn-accent);
  }
  &.btn-ghost-black {
    @include button-ghost($btn-black);
  }
  &.btn-ghost-green {
    @include button-ghost($btn-green);
  }
  &.btn-ghost-blue-light {
    @include button-ghost($btn-blue-light);
  }
  &.btn-ghost-blue {
    @include button-ghost($btn-blue);
  }
  &.btn-ghost-violet {
    @include button-ghost($btn-violet);
  }
  &.btn-ghost-red {
    @include button-ghost($btn-red);
  }
  &.btn-ghost-orange {
    @include button-ghost($btn-orange);
  }
  &.btn-ghost-yellow {
    @include button-ghost($btn-yellow);
  }

  &.btn-float-accent {
    @include button-float($btn-accent);
  }
  &.btn-float-black {
    @include button-float($btn-black);
  }
  &.btn-float-green {
    @include button-float($btn-green);
  }
  &.btn-float-blue-light {
    @include button-float($btn-blue-light);
  }
  &.btn-float-blue {
    @include button-float($btn-blue);
  }
  &.btn-float-violet {
    @include button-float($btn-violet);
  }
  &.btn-float-red {
    @include button-float($btn-red);
  }
  &.btn-float-orange {
    @include button-float($btn-orange);
  }
  &.btn-float-yellow {
    @include button-float($btn-yellow);
  }
}

/* Size / Fitting / Rounded / Icon Sizing
   ========================================================================== */
.btn {
  &.btn-xs {
    height: 30px;
    font-size: 13px;
    padding: 0 9px;

    &.btn-rounded {
      border-radius: 15px;
    }

    .icon, .icon svg {
      max-height: 16px;
      max-width: 16px;
    }
  }

  &.btn-s {
    height: 36px;
    font-size: 14px;
    padding: 0 12.5px;

    &.btn-rounded {
      border-radius: 18px;
    }

    .icon, .icon svg {
      max-height: 16px;
      max-width: 16px;
    }
  }

  &.btn-m {
    height: 42px;
    font-size: 14px;
    padding: 0 12px;

    &.btn-rounded {
      border-radius: 21px;
    }

    .icon, .icon svg {
      max-height: 16px;
      max-width: 16px;
    }
  }

  &.btn-l {
    height: 48px;
    font-size: 16px;
    padding: 0 12px;

    &.btn-rounded {
      border-radius: 24px;
    }

    .icon, .icon svg {
      max-height: 20px;
      max-width: 20px;
    }
  }

  &.btn-xl {
    height: 54px;
    font-size: 16px;
    padding: 0 12px;

    &.btn-rounded {
      border-radius: 27px;
    }

    .icon, .icon svg {
      max-height: 20px;
      max-width: 20px;
    }
  }

  &.btn-xxl {
    height: 60px;
    font-size: 18px;
    padding: 0 18.5px;

    &.btn-rounded {
      border-radius: 30px;
    }

    .icon, .icon svg {
      max-height: 20px;
      max-width: 20px;
    }
  }

  &.btn-borderRadius-6px {
    border-radius: 6px;
  }

  &.btn-fullwidth {
    width: 100%;
  }
}

/* Font Weight
   ========================================================================== */
.btn {
  &.btn-lightweight {
    font-weight: normal;
  }

  &.btn-middleweight {
    font-weight: 500;
  }

  &.btn-heavyweight {
    font-weight: 900;
  }
}

/* Content
   ========================================================================== */
.btn {

  .icon, .button-label {
    display: flex;
  }

  &.btn-lone-icon {
    border-radius: 50%;

    &.btn-xs {
      width: 30px;
      padding: 7px;
    }

    &.btn-s {
      width: 36px;
      padding: 10px;
    }

    &.btn-m {
      width: 42px;
      padding: 13px;
    }

    &.btn-l {
      width: 48px;
      padding: 14px;
    }

    &.btn-xl {
      width: 54px;
      padding: 17px;
    }

    &.btn-xxl {
      width: 60px;
      padding: 20px;
    }
  }

  &.btn-xs {

    .button-label-icon-left-center, .button-label-icon-right-center {
      margin: 0;
    }

    .button-label-icon-left-justified {
      margin: 0 0 0 auto;
    }

    .button-label-icon-right-justified {
      margin: 0 auto 0 0;
    }

    .icon-left-center, .icon-left-justified {
      margin: 0 5px 0 0;
    }

    .icon-right-center, .icon-right-justified {
      margin: 0 0 0 5px;
    }
  }

  &.btn-s {

    .button-label-icon-left-center, .button-label-icon-right-center {
      margin: 0;
    }

    .button-label-icon-left-justified {
      margin: 0 0 0 auto;
    }

    .button-label-icon-right-justified {
      margin: 0 auto 0 0;
    }

    .icon-left-center, .icon-left-justified {
      margin: 0 8.5px 0 0;
    }

    .icon-right-center, .icon-right-justified {
      margin: 0 0 0 8.5px;
    }
  }

  &.btn-m {

    .button-label-icon-left-center, .button-label-icon-right-center {
      margin: 0;
    }

    .button-label-icon-left-justified {
      margin: 0 0 0 auto;
    }

    .button-label-icon-right-justified {
      margin: 0 auto 0 0;
    }

    .icon-left-center, .icon-left-justified {
      margin: 0 11px 0 0;
    }

    .icon-right-center, .icon-right-justified {
      margin: 0 0 0 11px;
    }
  }

  &.btn-l {

    .button-label-icon-left-center, .button-label-icon-right-center {
      margin: 0;
    }

    .button-label-icon-left-justified {
      margin: 0 0 0 auto;
    }

    .button-label-icon-right-justified {
      margin: 0 auto 0 0;
    }

    .icon-left-center, .icon-left-justified {
      margin: 0 9px 0 0;
    }

    .icon-right-center, .icon-right-justified {
      margin: 0 0 0 9px;
    }
  }

  &.btn-xl {

    .button-label-icon-left-center, .button-label-icon-right-center {
      margin: 0;
    }

    .button-label-icon-left-justified {
      margin: 0 0 0 auto;
    }

    .button-label-icon-right-justified {
      margin: 0 auto 0 0;
    }

    .icon-left-center, .icon-left-justified {
      margin: 0 9.5px 0 0;
    }

    .icon-right-center, .icon-right-justified {
      margin: 0 0 0 9.5px;
    }
  }

  &.btn-xxl {

    .button-label-icon-left-center, .button-label-icon-right-center {
      margin: 0;
    }

    .button-label-icon-left-justified {
      margin: 0 0 0 auto;
    }

    .button-label-icon-right-justified {
      margin: 0 auto 0 0;
    }

    .icon-left-center, .icon-left-justified {
      margin: 0 15.5px 0 0;
    }

    .icon-right-center, .icon-right-justified {
      margin: 0 0 0 15.5px;
    }
  }
}

/* Button Group
   ========================================================================== */
.btngroup {
  display: inline-flex;

  .btngroup-list {
    list-style-type: none;
    margin: 0;
    padding: 0;
    display: inline-flex;

    .btngroup-item {
      display: inline-flex;
      align-items: center;
    }
  }

  &.btngroup-segmented {

    .btngroup-item {

      &#btngroup-firstitem {

        .btn {
          border-radius: 4px 0 0 4px;
        }

        &.btn-borderRadius-6px {
          border-radius: 6px 0 0 6px;
        }

        &.btn-rounded {
          .btn {

            &.btn-xs {
              border-radius: 15px 0 0 15px;
            }
            &.btn-s {
              border-radius: 18px 0 0 18px;
            }
            &.btn-m {
              border-radius: 21px 0 0 21px;
            }
            &.btn-l {
              border-radius: 24px 0 0 24px;
            }
            &.btn-xl {
              border-radius: 27px 0 0 27px;
            }
            &.btn-xxl {
              border-radius: 30px 0 0 30px;
            }
          }
        }
      }

      &#btngroup-lastitem {

        .btn {
          border-radius: 0 4px 4px 0;
        }

        &.btn-borderRadius-6px {
          border-radius: 0 6px 6px 0;
        }

        &.btn-rounded {

          .btn {

            &.btn-xs {
              border-radius: 0 15px 15px 0;
            }
            &.btn-s {
              border-radius: 0 18px 18px 0;
            }
            &.btn-m {
              border-radius: 0 21px 21px 0;
            }
            &.btn-l {
              border-radius: 0 24px 24px 0;
            }
            &.btn-xl {
              border-radius: 0 27px 27px 0;
            }
            &.btn-xxl {
              border-radius: 0 30px 30px 0;
            }
          }
        }
      }

      &[id^='btngroup-miditem-'] .btn {
        border-radius: 0;
      }
    }
  }
}
