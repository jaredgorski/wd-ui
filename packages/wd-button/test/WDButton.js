import WDButton from '../src/WDButton';

let button;

describe('WDButton', () => {
  afterEach(() => {
    if (button) {
      button.dispose();
    }
  });

  it('should generate the default markup', () => {
    button = new WDButton();

    expect(button).toMatchSnapshot();
  });
});
