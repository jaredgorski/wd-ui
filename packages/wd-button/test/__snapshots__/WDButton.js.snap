// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WDButton should generate the default markup 1`] = `<div id="example">Hello World</div>`;
